llvm/lib/CodeGen/DeadMachineInstructionElim.cpp

```diff
@@ -68,7 +68,7 @@ DeadMachineInstructionElimPass::run(MachineFunction &MF,
                                     MachineFunctionAnalysisManager &) {
   if (!DeadMachineInstructionElimImpl().runImpl(MF))
     return PreservedAnalyses::all();
-  PreservedAnalyses PA;
+  PreservedAnalyses PA = getMachineFunctionPassPreservedAnalyses();
```

Comment:
we should have an assert in the function -> machine function adaptor that function analyses were not invalidated, but can do that in a followup patch